src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/BrevPeriodeContext.kt

```diff
@@ -276,20 +278,36 @@ class BrevPeriodeContext(
                 val nullableSanitybegrunnelse = vedtakBegrunnelse.begrunnelse.tilSanityBegrunnelse(sanityBegrunnelser)
                 nullableSanitybegrunnelse?.let { Pair(vedtakBegrunnelse.begrunnelse, it) }
             }.map { (begrunnelse, sanityBegrunnelse) ->
-                val personerMedVilkårSomPasserBegrunnelse = begrunnelserForPeriodeContext
-                    .hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(
+
+                val relevantePersoner = when (begrunnelse.begrunnelseType) {
+                    BegrunnelseType.ETTER_ENDRET_UTBETALING -> begrunnelserForPeriodeContext.hentPersonerMedEndretUtbetalingerSomPasserMedVedtaksperiode(
+                        sanityBegrunnelse
+                    )
+
+                    else -> begrunnelserForPeriodeContext.hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(
                         begrunnelse = begrunnelse,
                         sanityBegrunnelse = sanityBegrunnelse
                     )
+                }
 
                 val antallTimerBarnehageplass =
-                    hentAntallTimerBarnehageplassTekst(personerMedVilkårSomPasserBegrunnelse)
+                    hentAntallTimerBarnehageplassTekst(relevantePersoner)
+
+                val relevanteEndringsperioderForBegrunnelse = begrunnelse.hentRelevanteEndringsperioderForBegrunnelse(
+                    sanityBegrunnelse = sanityBegrunnelse,
+                    endretUtbetalingAndeler = andelTilkjentYtelserMedEndreteUtbetalinger.flatMap { it.endreteUtbetalinger },
+                    vedtaksperiode = utvidetVedtaksperiodeMedBegrunnelser
+                )
 
-                val gjelderSøker = personerMedVilkårSomPasserBegrunnelse.any { it.type == PersonType.SØKER }
+                val søknadstidspunkt =
+                    relevanteEndringsperioderForBegrunnelse.sortedBy { it.søknadstidspunkt }
+                        .firstOrNull()?.søknadstidspunkt
```

Comment:
Er ikke dette det samme som `relevanteEndringsperioderForBegrunnelse.minOfOrNull { it.søknadstidspunkt }`?

Reply:
Stemmer, men minByOrNull tar ikke inn nullables. 😞

Reply:
Aah sett 😞